### PR TITLE
fix static linking for tests and extras, remove unneeded -static flag

### PR DESCRIPTION
**Pull request by schamane**
Hi,

on macosx you cant staticaly link with "-static" flag for binaries. So it should be removed.
Please review my pool request for this :D

### CODE REVIEW

File: extras/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,11 @@ function(rwkv_add_extra source)
 	get_filename_component(EXTRA_TARGET ${source} NAME_WE)
 	add_executable(rwkv_${EXTRA_TARGET} ${source})
 	target_link_libraries(rwkv_${EXTRA_TARGET} PRIVATE ggml rwkv)
+	if (RWKV_STATIC)
+		get_target_property(target_LINK_OPTIONS rwkv_${EXTRA_TARGET} LINK_OPTIONS)
+    	list(REMOVE_ITEM target_LINK_OPTIONS "-static")
+    	set_target_properties(rwkv_${EXTRA_TARGET} PROPERTIES LINK_OPTIONS "${target_LINK_OPTIONS}")
```
Comment (saharNooby) [inline extras/CMakeLists.txt:8]:
```suggestion
    	    set_target_properties(rwkv_${EXTRA_TARGET} PROPERTIES LINK_OPTIONS "${target_LINK_OPTIONS}")
```
And above

File: extras/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,11 @@ function(rwkv_add_extra source)
 	get_filename_component(EXTRA_TARGET ${source} NAME_WE)
 	add_executable(rwkv_${EXTRA_TARGET} ${source})
 	target_link_libraries(rwkv_${EXTRA_TARGET} PRIVATE ggml rwkv)
+	if (RWKV_STATIC)
+		get_target_property(target_LINK_OPTIONS rwkv_${EXTRA_TARGET} LINK_OPTIONS)
```
Comment (saharNooby) [inline extras/CMakeLists.txt:6]: Why there is `rwkv_` prefix in `rwkv_${EXTRA_TARGET}`, but no such prefix in [tests/CMakeLists.txt](https://github.com/saharNooby/rwkv.cpp/pull/98/files#diff-4461c617ceae0c7e0206622aacdf555aedd62eb129c2efb74c84fa1567bcbe0d)?

**Comment by saharNooby** [comment]
Do I understand correctly that the added code just removes "-static" option from compiler options when compiling extras and tests?

CC @LoganDark just in case -- he is more familliar with C/C++ :)

File: extras/CMakeLists.txt (hunk 2 above)
Comment (schamane) [reply extras/CMakeLists.txt:6]: I have no idea why there is prefix. Its looks like you defined it in CMakeLists :D

**Comment by schamane** [comment]
Yes, at least on macosx you dont need "-static" flag to build binaries, it event worse. It break your build.

https://leanprover-community.github.io/archive/stream/113488-general/topic/libgmp.20dependency.html

image

**Comment by schamane** [comment]
Thats true @LoganDark, but on mac build process crash if you try to build statically tests and extras...

Here is the way to reproduce:

```bash
mkdir build && cd build
cmake -DRWKV_BUILD_SHARED_LIBRARY=OFF -DRWKV_STATIC=ON ..
cmake --build . --config Release
```

Output:

```text
4 warnings generated.
[ 23%] Linking CXX static library librwkv.a
[ 23%] Built target rwkv
[ 30%] Building C object tests/CMakeFiles/test_ggml_basics.dir/test_ggml_basics.c.o
[ 38%] Linking CXX executable ../bin/test_ggml_basics
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/test_ggml_basics] Error 1
make[1]: *** [tests/CMakeFiles/test_ggml_basics.dir/all] Error 2
```

**Comment by LoganDark** [comment]
> Thats true @LoganDark, but on mac build process crash if you try to build statically tests and extras...

does removing `-static` still allow you to statically link with rwkv.cpp?

File: extras/CMakeLists.txt [hunk 3]
```diff
@@ -2,6 +2,11 @@ function(rwkv_add_extra source)
...
+    	list(REMOVE_ITEM target_LINK_OPTIONS "-static")
+    	    set_target_properties(rwkv_${EXTRA_TARGET} PROPERTIES LINK_OPTIONS "${target_LINK_OPTIONS}")
```
Comment (LoganDark) [inline extras/CMakeLists.txt:8]: use tabs for indentation or convert the whole file to use spaces

**Review by LoganDark — CHANGES_REQUESTED** [review]

**Review by LoganDark — APPROVED** [review]
LGTM, doesn't look like these need to be statically linked on any platform

File: extras/CMakeLists.txt (hunk 3 above)
Comment (schamane) [reply extras/CMakeLists.txt]: ok, i will do. It comes somehow wrong over sugestions

Comment (LoganDark) [reply extras/CMakeLists.txt]: it's fixed now

Comment (schamane) [reply extras/CMakeLists.txt]: @LoganDark @saharNooby no idea why, extras use tabs, and tests use spaces :D

**Comment by schamane** [comment]
Done

File: extras/CMakeLists.txt (hunk 3 above)
Comment (LoganDark) [reply extras/CMakeLists.txt]: because I used tabs when contributing extras, it was an accident

**Review by saharNooby — APPROVED** [review]

**Pull request merged**